### PR TITLE
Add instructions for linking to the dependent repositories

### PR DESCRIPTION
**Pull request by BillWagner**
The dependent repositories for the language specification requires special handling for the links to resolve correctly by the build system.

See https://github.com/dotnet/docs/pull/8221#issuecomment-428943541 for details.

/cc @Mackiovello

### CODE REVIEW

File: styleguide/template.md
Comment (Mackiovello) [inline styleguide/template.md]: Typo: "i'ts" => "it's"

**Review by ErlendLandro — COMMENTED** [review]
Thank you for adding these instructions!

Found one typo

File: styleguide/template.md
Comment (pkulikov) [inline styleguide/template.md]: I see that it's like other examples in this document, but sadly, those examples do not help a lot. Those examples are rendered as links in the GitHub web UI. However, as the one who edits a document, I prefer to see *how* to produce a link.

Should we put examples in backticks: 
>- Example: `[C# Query Expressions](~/_csharplang/spec/expressions.md#query-expressions)`

OR combine both source text and produced link:
>- Example: `[C# Query Expressions](~/_csharplang/spec/expressions.md#query-expressions)` produces [C# Query Expressions](~/_csharplang/spec/expressions.md#query-expressions)

File: styleguide/template.md
Comment (pkulikov) [inline styleguide/template.md]: Links to the repositories don't work in the preview. Should those be full links:
https://github.com/dotnet/csharplang
https://github.com/dotnet/vblang

**Review by rpetrusha — APPROVED** [review]
This looks good, @BillWagner. You can merge when you're ready.

File: styleguide/template.md
Comment (BillWagner) [reply styleguide/template.md]: I just saw this after hitting the merge button.  

I'll make a new PR for this and your earlier comment regarding the format.

Comment (BillWagner) [reply styleguide/template.md]: See #8298

File: styleguide/template.md
Comment (BillWagner) [reply styleguide/template.md]: See #8298

**Pull request merged**
